### ISSUE TITLE
Cant pay through channel untill reconnect with this peer

### ISSUE THREAD

**Issue by Overtorment** [lightningnetwork/lnd#4270]
### Background

trying to pay ACINQ invoice, direct channel, channel is active, payment hangs:

```
overtorment@lnd2:~/go/src/github.com$ lncli trackpayment 2155f5a489f0a3bb6286b21ae0e7ac93ef27d34449a0e1a49ced26e0963b2363                                                                                           
                           +------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT           | ROUTE |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
| IN_FLIGHT  |        0.028 |            - | 1            | 0   |   629783 | 691463071535595520 | ACINQ |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
Amount + fee:   0 + 0 sat
Payment hash:   2155f5a489f0a3bb6286b21ae0e7ac93ef27d34449a0e1a49ced26e0963b2363
Payment status: IN_FLIGHT
```

payment was sent with `timeout_seconds` = 6.
don't have exact response from lnd, but retrying this exact bolt11 produces:

```
overtorment@lnd2:~/go/src/github.com$ lncli sendpayment --pay_req       lnbc10n1p0t0ewspp5y92ltfyf7z3mkc5xkgdwpeavj0hj056yfxswrfyua5nwp93myd3sdqcv93kxetsw3skucm9yp6x2um5cqp2xqrrss9qy9qsqsp5tpy56rqsvst3vm823rqgysf
urc3se8wy65hmcln3rmzl4vv0tz3qg38mdtf6ln5qwqynqh08r7wlp3d9tr7c9gnw4zy7e3uq2vxz5q7xdenx8zl0vn40psx5zslczc7l3rvdkr69zkmxyr0x9ke0regc3aqq66ahzc
Payment hash: 2155f5a489f0a3bb6286b21ae0e7ac93ef27d34449a0e1a49ced26e0963b2363
Description: acceptance test
Amount (in satoshis): 1
Fee limit (in satoshis): 1
Destination: 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f
Confirm payment (yes/no): yes
[lncli] rpc error: code = AlreadyExists desc = payment is in transition
```

then doing reonnect :

```
overtorment@lnd2:~/go/src/github.com$ lncli disconnect 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f
{

}
overtorment@lnd2:~/go/src/github.com$ lncli connect 03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f@34.239.230.56:9735
{

}
```

and payment went through:

```
overtorment@lnd2:~/go/src/github.com$ lncli trackpayment 2155f5a489f0a3bb6286b21ae0e7ac93ef27d34449a0e1a49ced26e0963b2363                                                                                                                      +------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
| HTLC_STATE | ATTEMPT_TIME | RESOLVE_TIME | RECEIVER_AMT | FEE | TIMELOCK | CHAN_OUT           | ROUTE |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
| SUCCEEDED  |        0.028 |      131.683 | 1            | 0   |   629783 | 691463071535595520 | ACINQ |
+------------+--------------+--------------+--------------+-----+----------+--------------------+-------+
Amount + fee:   1 + 0 sat
Payment hash:   2155f5a489f0a3bb6286b21ae0e7ac93ef27d34449a0e1a49ced26e0963b2363
Payment status: SUCCEEDED, preimage: 02605372d276497f41c3f2c9f1e69acfd32b3b02403c92d1974befe1a280befe
```

### Your environment

* version of `lnd` 0.10
* which operating system (`uname -a` on *Nix) ubuntu 18
* version of `btcd`, `bitcoind`, or other backend neutrino
sendPaymentV2 through nodejs grpc

### Steps to reproduce

^^^

### Expected behaviour

payments go well through direct active channels

### Actual behaviour

stuck in direct channel

probably related to https://github.com/lightningnetwork/lnd/issues/4267
probably even the same issue

**Comment by Overtorment** [comment]
[acinq.full.log.zip](https://github.com/lightningnetwork/lnd/files/4605588/acinq.full.log.zip)
attaching full debug log  for that hour, no greps. reconnect that actually helped happened around `11:32`

**Comment by Roasbeef** [comment]
Fixed by #4183.
